### PR TITLE
More details on source licenses

### PR DESCRIPTION
**Pull request by jlblancoc**
cc: https://github.com/ros/rosdistro/pull/35980

### CODE REVIEW

**Comment by jlblancoc** [comment]
Please, @nuclearsandwich let me know if this changes to GTSAM's package.xml are ok?

File: package.xml [hunk 1]
```diff
@@ -6,8 +6,17 @@
   <description>gtsam</description>
 
   <maintainer email="gtsam@lists.gatech.edu">Frank Dellaert</maintainer>
+
+  <!-- Main License for GTSAM sources -->
   <license>BSD</license>
```
Comment (nuclearsandwich) [inline package.xml:11]:
```suggestion
  <license>BSD-3-Clause</license>
```

Although `BSD` usually indicates the 3 Clause BSD license within the ROS community I think explicitly using the SPDX.org identifier for it is an improvement I recommend for every ROS package.

File: package.xml [hunk 2]
```diff
@@ -6,8 +6,17 @@
...
-  
+  <!-- 3rd party libraries -->
+  <license file="gtsam/3rdparty/CCOLAMD/Doc/License.txt">BSD</license>
```
Comment (nuclearsandwich) [inline package.xml:13]:
```suggestion
  <license file="gtsam/3rdparty/CCOLAMD/Doc/License.txt">BSD-3-Clause</license>
```

Use full SPDX.org license identifier for 3 clause BSD.

File: package.xml [hunk 3]
```diff
@@ -6,8 +6,17 @@
...
+  <license file="gtsam/3rdparty/ceres/jet.h">BSD</license>
```
Comment (nuclearsandwich) [inline package.xml:14]:
```suggestion
  <license file="gtsam/3rdparty/ceres/jet.h">BSD-3-Clause</license>
```

Use full SPDX.org license identifier for 3 clause BSD.

File: package.xml [hunk 4]
```diff
@@ -6,8 +6,17 @@
...
+  <license file="gtsam/3rdparty/Eigen/COPYING.README">MPL2</license>
```
Comment (nuclearsandwich) [inline package.xml:15]:
```suggestion
  <license file="gtsam/3rdparty/Eigen/COPYING.README">MPL-2.0</license>
```

Use the full [SPDX.org](https://spdx.org/licenses/) license identifier for MPL 2.0.

File: package.xml [hunk 5]
```diff
@@ -6,8 +6,17 @@
...
+  <license file="gtsam/3rdparty/GeographicLib/LICENSE.txt">MIT</license>
+  <license file="gtsam/3rdparty/metis/LICENSE.txt">Apache-2.0</license>
+  <license file="gtsam/3rdparty/Spectra/GenEigsSolver.h">MPL2</license>
```
Comment (nuclearsandwich) [inline package.xml:18]:
```suggestion
  <license file="gtsam/3rdparty/Spectra/GenEigsSolver.h">MPL-2.0</license>
```

Use the full [SPDX.org](https://spdx.org/licenses/) license identifier for MPL 2.0.

**Review by nuclearsandwich — APPROVED** [review]
@jlblancoc thanks for proposing these changes. This kind of license info is exactly what we recommend as the best current practice for packages which contain files under multiple licenses.

I've added a few suggestions that switch the short names for some licenses to the SPDX.org standard license identifiers. Especially for ambiguous identifiers like `BSD` which can refer to different variations using an SPDX identifier can dramatically assist license review automation tools.

But even if those suggestions aren't taken for some reason this is still a big improvement!

File: package.xml (hunk 1 above)
Comment (jlblancoc) [reply package.xml:11]: exactly, I saw somewhere in the ROS docs that BSD = 3-clause BSD. Anyway, as long as your more explicit names do not raise any warning in any ROS tool or linter, I also like them more. 
Thanks!

**Comment by jlblancoc** [comment]
@dellaert I cancelled the CI builds to avoid wasting resources. Feel free of merging when possible, it seems ok now!

**Review by dellaert — APPROVED** [review]

**Comment by dellaert** [comment]
Still needs to pass a required CI so I'll fire that off and merge. Thanks!!!!

**Pull request merged**